test-runner.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+################################################################################
+# Ensure our environment variables are set
+################################################################################
+if [ "$PHP_VERSION" == "" ]; then
+  echo "You must define a \$PHP_VERSION environment variable"
+  exit 1
+fi
+
+if [ "$IMAGE_FLAVOUR" == "" ]; then
+  echo "You must define a \$IMAGE_FLAVOUR environment variable"
+  exit 1
+fi
+
+if [ "$DOCKER_IP" == "" ]; then
+  DOCKER_IP="$(dinghy ip)"
+fi
+
+################################################################################
+# Set our variables
+################################################################################
+IMAGE_NAME="test-$PHP_VERSION-$IMAGE_FLAVOUR"
+TEST_DIR="$(pwd)/test"
+TEST_OK_STRING="TEST OK"
+TEST_PORT="8888"
+
+################################################################################
+# Kick off with some debug
+################################################################################
+echo ""
+echo "Test Runner Configuration:"
+echo ""
+echo "PHP Version: $PHP_VERSION"
+echo "Image Flavour: $IMAGE_FLAVOUR"
+
+################################################################################
+# Pull published image down so we can try and reuse layers
+################################################################################
+echo ""
+echo "Pulling published images for layer cache.."
+echo ""
+docker pull meanbee/magento:${PHP_VERSION}-${IMAGE_FLAVOUR}
+
+################################################################################
+# Build our images
+################################################################################
+echo ""
+echo "Building image.."
+echo ""
+cd $PHP_VERSION/$IMAGE_FLAVOUR && docker build -t $IMAGE_NAME . || exit 1
+cd ../..
+
+################################################################################
+# Output the PHP version running in the image
+################################################################################
+echo ""
+echo "Image PHP Version:"
+echo ""
+docker run $IMAGE_NAME php --version || exit 1
```

Comment:
I'm not very familiar with Travis and how it uses instances, but we should probably add `--rm` to `docker run` commands to save resources